cachepool/cachepool.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python2
+# coding: utf-8
+
+import copy
+import Queue
+import logging
+
+logger = logging.getLogger(__name__)
+
+class CachePoolError(Exception): pass
+class CachePoolGeneratorError(CachePoolError): pass
+class CachePoolCloseCallbackError(CachePoolError): pass
+
+class CachePoolReuseDeciderError(CachePoolError): pass
+
+class CachePool(object):
+
+    def __init__(self,
+                    generator,
+                    generator_args=[],
+                    generator_argkw={},
+                    close_callback=None,
+                    pool_size=1024):
+
+        self.generator = generator
+        self.generator_args = copy.deepcopy(generator_args)
```

Comment:
There might be a side effect of using `copy.deepcopy` with the arguments.
An argument might be a class instance, a file descriptor or a lock.

Leave to the caller with the decision whether or not to duplicate the arguments.